source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -407,16 +407,38 @@ export class CardDesignerSurface {
             peer.onChanged = (sender: DesignerPeers.DesignerPeer, updatePropertySheet: boolean) => { this.peerChanged(sender, updatePropertySheet); };
             peer.onPeerRemoved = (sender: DesignerPeers.DesignerPeer) => { this.peerRemoved(sender); };
             peer.onPeerAdded = (sender: DesignerPeers.DesignerPeer, newPeer: DesignerPeers.DesignerPeer) => {
-                this.addPeer(newPeer);
+                if (newPeer.insertAfterNeighbor) {
+                    this.addPeer(newPeer, this.getPeerHTMLNeighbor(newPeer));
+                } else {
+                    this.addPeer(newPeer);
+                }
+
+                newPeer.insertAfterNeighbor = false;
+                
                 this.updateLayout();
             };
             peer.onStartDrag = (sender: DesignerPeers.DesignerPeer) => { this.startDrag(sender); }
             peer.onEndDrag = (sender: DesignerPeers.DesignerPeer) => { this.endDrag(false); }
 
-            peer.addElementsToDesignerSurface(this._designerSurface);
+            peer.addElementsToDesignerSurface(this._designerSurface, neighbor);
 
+            // If we process children, we need to make sure we track the child elements here as well
+            // Test scenario: when we add a row to a table, the TableRow is a container, so the children of the container must be added to the right spot as well
+            // TODO: it might be worth checking the insertAfterNeighbor variable here as well?
+            let currentNeighbor = peer;
             for (let i = 0; i < peer.getChildCount(); i++) {
-                this.addPeer(peer.getChildAt(i));
+                let currentPeer = peer.getChildAt(i);
```

Comment:
here and elsewhere in this file

```suggestion
                const currentPeer = peer.getChildAt(i);
```

a best practice (that wasn't being followed before) is to use `const` when you aren't assigning a variable again in the same method -- it's a code correctness thing that will prevent accidental assignments in the future (and establishes author intent around how the variable is supposed to be used). I should note that unlike other languages, JS doesn't enforce constness beyond just assignments, so the following works without error (and wouldn't compile in the equivalent C++):

```javascript
const someObj = { a: "string A", b: "string B" };
someObj.a = "this is perfectly legal";
```

Reply:
Good call out! I'll go through and update these :)